lib/attrsets.nix

```
@@ -651,6 +651,20 @@ rec {
     f:
     listToAttrs (map (n: nameValuePair n (f n)) names);
 
+  /* Generate an attribute set by mapping over a list of items
+     Taking two functions, one to apply for the name and one to
+     apply for the value of the attribute set.
+
+    Example:
+       genAttrs' [ {a = 2; b.c = "a" } {a = 1; b.c = "b"} ]
+          (i: i.b.c) (i: i.a)
```

Don't overload `a` and `b`, e.g.:

```suggestion
       genAttrs' [ {p = 2; q.r = "a" } {p = 1; q.r = "b"} ]
          (i: i.q.r) (i: i.p)
```